File: problem-solving/lc-1-easy-two-sum/1-brute-force.go
/*
Problem: https://leetcode.com/problems/two-sum/

Complexity
----------
Time: O(N^2)
Space: O(1)
*/

func twoSum(nums []int, target int) []int {
    for i := 0; i < len(nums); i++ {
        for j := i + 1; j < len(nums); j++ {
            if nums[i] + nums[j] == target {
                return []int{i, j}
            }
        }
    }
    return []int{}
}

File: problem-solving/lc-377-medium-combination-sum-iv/1-brute-force.py
"""
https://leetcode.com/problems/combination-sum-iv

Complexity
----------
Time: 
Space: 
"""

class Solution:
    def combinationSum4(self, nums: List[int], target: int) -> int:
        if target == 0:
            return 1
        count = 0
        for n in nums:
            if n <= target:
                count += self.combinationSum4(nums, target - n)
        return count
